support/config/deployment.go

```diff
@@ -145,28 +153,41 @@ func colocationLabelValue(hcp *hyperv1.HostedControlPlane) string {
 	return clusterKey(hcp)
 }
 
+const podHashLabel = "hypershift.openshift.io/pod-template-hash"
+
 // setMultizoneSpread sets PodAntiAffinity with corev1.LabelTopologyZone as the topology key for a given set of labels.
 // This is useful to e.g ensure pods are spread across availavility zones.
-func (c *DeploymentConfig) setMultizoneSpread(labels map[string]string) {
-	if labels == nil {
+func (c *DeploymentConfig) setMultizoneSpread(pod *corev1.PodTemplateSpec) {
+	if !c.setDefaults || c.Replicas <= 1 {
 		return
```

Comment:
do we really need any check? since we are using a unique hash per Deployment I think could apply this unconditionally now.